test/loader/serializeInstructionTest.cpp

```diff
@@ -1126,4 +1242,360 @@ TEST(SerializeInstructionTest, SerializeDotProductInstruction) {
+
+  Conf.addProposal(WasmEdge::Proposal::GC);
+  StructNew.getTargetIndex() = 0xFFFFFFFFU;
+  Instructions = {StructNew, End};
+  Output = {};
+  EXPECT_TRUE(Ser.serializeSection(createCodeSec(Instructions), Output));
+  Expected = {
+      0x0AU,                             // Code section
+      0x0BU,                             // Content size = 11
+      0x01U,                             // Vector length = 1
+      0x09U,                             // Code segment size = 9
+      0x00U,                             // Local vec(0)
+      0xFBU, 0x00U,                      // OpCode Struct__new.
+      0xFFU, 0xFFU, 0xFFU, 0xFFU, 0x0FU, // Type index.
+      0x0BU                              // Expression End.
+  };
+  EXPECT_EQ(Output, Expected);
+
+  StructNewDefault.getTargetIndex() = 0xFFFFFFFFU;
+  Instructions = {StructNewDefault, End};
+  Output = {};
+  EXPECT_TRUE(Ser.serializeSection(createCodeSec(Instructions), Output));
+  Expected[6] = 0x01U; // OpCode Struct__new_default.
+  EXPECT_EQ(Output, Expected);
+
+  StructGet.getSourceIndex() = 0x05U;
+  StructGet.getTargetIndex() = 0xFFFFFFFFU;
+  Instructions = {StructGet, End};
+  Output = {};
+  EXPECT_TRUE(Ser.serializeSection(createCodeSec(Instructions), Output));
+  Expected = {
+      0x0AU,                             // Code section
+      0x0CU,                             // Content size = 12
+      0x01U,                             // Vector length = 1
+      0x0AU,                             // Code segment size = 10
+      0x00U,                             // Local vec(0)
+      0xFBU, 0x02U,                      // OpCode Struct__get.
+      0xFFU, 0xFFU, 0xFFU, 0xFFU, 0x0FU, // Type index.
+      0x05U,                             // Source index.
+      0x0BU                              // Expression End.
+  };
+  EXPECT_EQ(Output, Expected);
+
+  StructGetS.getSourceIndex() = 0x05U;
+  StructGetS.getTargetIndex() = 0xFFFFFFFFU;
+  Instructions = {StructGetS, End};
+  Output = {};
+  EXPECT_TRUE(Ser.serializeSection(createCodeSec(Instructions), Output));
+  Expected[6] = 0x03U; // OpCode Struct__get_s.
+  EXPECT_EQ(Output, Expected);
+
+  StructGetU.getSourceIndex() = 0x05U;
+  StructGetU.getTargetIndex() = 0xFFFFFFFFU;
+  Instructions = {StructGetU, End};
+  Output = {};
+  EXPECT_TRUE(Ser.serializeSection(createCodeSec(Instructions), Output));
+  Expected[6] = 0x04U; // OpCode Struct__get_u.
+  EXPECT_EQ(Output, Expected);
+
+  StructSet.getSourceIndex() = 0x05U;
+  StructSet.getTargetIndex() = 0xFFFFFFFFU;
+  Instructions = {StructSet, End};
+  Output = {};
+  EXPECT_TRUE(Ser.serializeSection(createCodeSec(Instructions), Output));
+  Expected[6] = 0x05U; // OpCode Struct__set.
+  EXPECT_EQ(Output, Expected);
+
+  Conf.removeProposal(WasmEdge::Proposal::GC);
+  Output = {};
+  EXPECT_FALSE(Ser.serializeSection(createCodeSec(Instructions), Output));
+}
+
+TEST(SerializeInstructionTest, SerializeArrayInstruction) {
```

Comment:
These instructions are also reference instrucitons. Combine them into the test set.